lib/sha256crypt.cpp

```diff
@@ -424,7 +433,7 @@ extern "C" char * sha256_crypt_r (const char *key, const char *salt, char *buffe
   /* Create byte sequence P.  */
   cp = p_bytes = (char *)alloca (key_len);
   for (cnt = key_len; cnt >= 32; cnt -= 32)
-    cp = (char *)mempcpy (cp, temp_result, 32);
```

Comment:
mempcpy is only available on macos

Reply:
True. But `mempcpy()` is not the same as `memcpy()` .
Reference:
https://man7.org/linux/man-pages/man3/mempcpy.3.html

"[...] instead of returning the value of dest it returns a pointer to the byte following the last written byte"

Reply:
Thanks for the hint is now fixed :)